### PR TITLE
[jvm-packages] update doc based on the latest changes

### PR DESCRIPTION
**Pull request by wbo4958**
To fix https://github.com/dmlc/xgboost/issues/10841

### CODE REVIEW

**Comment by wbo4958** [comment]
Hi @eordentlich, @trivialfis, would you help review it? Thx very much

File: doc/jvm/xgboost_spark_migration.rst [hunk 1]
```diff
@@ -63,73 +63,26 @@ Learning to rank using XGBoostRegressor has been replaced by a dedicated `XGBoos
...
-  // before 3.0
+
+  // before xgboost4j-spark 3.0
   val regressor = new XGBoostRegressor().setObjective("rank:ndcg")
 
-  // after 3.0
+  // after xgboost4j-spark 3.0
   val ranker = new XGBoostRanker()
 
-******************************
```
Comment (trivialfis) [inline doc/jvm/xgboost_spark_migration.rst:73]: Is this no longer needed?

Comment (wbo4958) [reply doc/jvm/xgboost_spark_migration.rst:73]: yes, I made https://github.com/dmlc/xgboost/pull/10845 to be compatible with old style.

**Comment by wbo4958** [comment]
Hi @trivialfis, Could you help review it?

**Review by trivialfis — APPROVED** [review]

**Pull request merged**